node/service/src/lib.rs

```diff
@@ -899,8 +917,7 @@ pub fn new_full<RuntimeApi, Executor, OverseerGenerator>(
 			if enable_approval_checking_voting_rule {
```

Comment:
We could remove this whole conditional, I think. The `RelayChainSelection` removes any need for the `ApprovalCheckingVotingRule`.

We could even remove the `enable_approval_checking_voting_rule` bool. As long as we only call `select_chain.connect_overseer_handler` on chains that actually need the `RelayChainSelection`, it will fall back to the `LongestChain` voting rule.

Reply:
> As long as we only call select_chain.connect_overseer_handler on chains that actually need the RelayChainSelection

We do it for chains that have overseer, which includes polkadot. Maybe I'm missing something, but not sure we can do this, essentially we need to know whether the `ParachainsHost` impl is dummy or not.

Reply:
> Still I have a concert wrt to polkadot and the reason why we have enable_approval_checking_voting_rule in the first place.

addressed in 4aebe3e, please take a look

Reply:
Yes, in Polkadot we'll need to avoid connecting the fork-choice rule to the overseer and then it'll use simple longest-chain.